### ISSUE TITLE
[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.NullPointerException

### ISSUE THREAD

**Issue by Jayonics** [microsoft/azure-tools-for-java#4045]
IntelliJ build version: 2019.3.3 IU-193.6494.35
OS: Windows 10
JDK: JetBrains s.r.o 11.0.5
Plugin version: 3.32.0-2019.3
Additional Info: None
Parent component: 
```
javax.swing.JRootPane[,3,0,777x489,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=java.awt.Dimension[width=777,height=489]]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.NullPointerException
	at com.microsoft.azure.hdinsight.spark.common.SparkLocalRunConfigurableModel.<init>(SparkLocalRunConfigurableModel.java:84)
	at com.microsoft.azure.hdinsight.spark.common.SparkBatchJobConfigurableModel.<init>(SparkBatchJobConfigurableModel.java:60)
	at com.microsoft.azure.hdinsight.spark.run.configuration.ArisSparkConfigurableModel.<init>(ArisSparkConfigurableModel.kt:29)
	at com.microsoft.azure.hdinsight.spark.run.configuration.ArisSparkConfigurationFactory.createTemplateConfiguration(ArisSparkConfigurationFactory.kt:39)
```

**Comment by Y4MES** [comment]
@wezhang I think it would be worth pointing out that this seems to be an issue with global data sources for the bundled databases plugin and not with Azure plugin for IntelliJ.

I was able to fix the issue every time by deleting the global data sources XML configuration within the %userprofile%/.IntelliJIDEA2019.3/config/options
[options.zip](https://github.com/microsoft/azure-tools-for-java/files/4270631/options.zip)
 directory. Then IntelliJ would boot correctly and whenever I would then add my database as a global data source instead of a local data source IntelliJ would crash upon loading any project after reopening intelliJ. Having the data source as a local (project specific) the problem would not occur.

This problem occurs on all of my devices (Laptop, Desktop and Server Desktop) when using the global configuration.

This may not be the correct place to post this issue so if you wouldn't mind redirecting me where to post this issue I would very much appreciate it.

The files are renamed as xml2 because I don't want the crash to reoccur

**Comment by wezhang** [comment]
Thanks @Jayonics ! Looks like the issue is caused by IntelliJ API `LocalFileSystem.getInstance().findFileByPath()` invoked by `Project::getBaseDir()` potentially returning `null`. I've merged a fix in the following release v3.33. Hope it can help you.

**Comment by Jayonics** [comment]
@wezhang thank you very much.
